### ISSUE TITLE
Silence Mode

### ISSUE THREAD

**Issue by UlisesGascon** [GoblinDBRocks/GoblinDB#13]
To be able to mute throw errors.

**Comment by CodingCarlos** [comment]
Something to have in mind while working on #18. Maybe create some error manager that, by default, throws an error, but has a flag to be permissive (console warn/error instead of throw) or to be 100% silent (still working even if error without logging nor throwing nothing).

**Comment by UlisesGascon** [comment]
I love the idea @CodingCarlos!

**Comment by Sediug** [comment]
I like the Carlos idea. I think that feature should be included as a configuration setting allowing the user of this package to set one of the following behaviours, throw the errors, console.warn / error or 100% silent.

**Comment by CodingCarlos** [comment]
Great to hear that, @Sediug! I'm moving the milestone to v0.1.0. I've just finished the first version of refactoring, and I'm checking if I've forgotten something before launch the PR to `dev`. 

I would like to know if @UlisesGascon and you talked about this yesterday, and how things went ;)

Anyway, if you take a look to the last version of the `refactor` branch, you'll see that I've created a "logger" folder into "lib", to make this issue easy to fix.

**Comment by Sediug** [comment]
Great! We didn't discuss it yesterday, neither the plugins. @UlisesGascon  was kind of busy, I didn't bring up the subject though. We could chat about it in Slack.

**Comment by Sediug** [comment]
Hi there!

I hope you're having a great weekend. I have a proposal but I'm working in the test because I think there're some errors. Ex:

![image](https://user-images.githubusercontent.com/11425316/33528474-3b6defd0-d861-11e7-9976-6ce412ac8f81.png)

![image](https://user-images.githubusercontent.com/11425316/33528533-5009e7d6-d862-11e7-92d7-47e21c3f71a1.png)

You're testing the method update expecting that to throw an error, but with the current method update it's not possible. I mean if the stored ambush function has a valid composition then calling merge with undefined as the second parameter will get you the same as the stored ambush has before.
```
_.merge(goblin.ambush[index], undefined) => goblin.ambush[index]
```
In other words, you want to test the new options, but you're always testing the overall new ambush func so if you don't provide options then the function will not be update or it'll be update with same composition as before but you'll never get an error. 

There're a problems like that in the test cases. From my point of view we've to options. Check the new options before to merge it with the old one's (and then merged) or remove this test.

The test was passing because you were importing the errors as var errors = '../lib/logger/errors'; instead of var errors = require('../lib/logger/errors');

Any ideas? Maybe I'm missing something, if thats the case I'm sorry xDD.

**Comment by Sediug** [comment]
Another problem on top of that, a few lines before a test case remove all the ambush functions so  the function in the mentioned case will not exist. This is easy to solve but just to have it in mind.

**Comment by Sediug** [comment]
What do you think? I didn't fix the test cases it's just the new implementation for the error manager or logger.

https://github.com/Sediug/GoblinDB/commit/42d474323e56f00f9b6299f6002b2c955b916210

**Comment by CodingCarlos** [comment]
Fixed on #31
